pkg/kubelet/cm/deviceplugin/manager_test.go

```diff
@@ -66,38 +87,41 @@ func TestDevicePluginReRegistration(t *testing.T) {
 		callbackCount++
 		callbackChan <- callbackCount
 	}
-	m, p1 := setup(t, devs, callback)
+	m, p1 := setup(t, socketName, pluginSocketName, devs, callback)
+	defer cleanup(t, m, p1)
+
 	p1.Register(socketName, testResourceName)
 	// Wait for the first callback to be issued.
-	<-callbackChan
-	// Wait till the endpoint is added to the manager.
-	for i := 0; i < 20; i++ {
-		if len(m.Devices()) > 0 {
-			break
-		}
-		time.Sleep(1)
+	select {
+	case <-callbackChan:
+	case <-time.After(5 * time.Second):
+		t.FailNow()
```

Comment:
cc @lichuqiang 
There is actually a race here that we can't just wait for callback before checking m.Devices(). Basically, with the current code, we can get a callback from endpoint.list() before the endpoint is added to the manager. With PR https://github.com/kubernetes/kubernetes/pull/52149 this race is gone so we can remove this wait. I would suggest we take this part of change out of this PR, wait after https://github.com/kubernetes/kubernetes/pull/52149 gets in, and then clean up this test.